drivers/adc/adc_mchp_xec_v2.c

```diff
@@ -304,45 +307,66 @@ static int adc_xec_init(const struct device *dev)
 	const struct adc_xec_config *const cfg = ADC_XEC_CONFIG(dev);
 	struct adc_xec_regs *adc_regs = ADC_XEC_REG_BASE(dev);
 	struct adc_xec_data *data = ADC_XEC_DATA(dev);
+	int ret;
+
+	ret = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret != 0) {
+		LOG_ERR("XEC ADC V2 pinctrl setup failed (%d)", ret);
+		return ret;
+	}
 
 	adc_regs->control_reg =  XEC_ADC_CTRL_ACTIVATE
 		| XEC_ADC_CTRL_POWER_SAVER_DIS
 		| XEC_ADC_CTRL_SINGLE_DONE_STATUS
 		| XEC_ADC_CTRL_REPEAT_DONE_STATUS;
 
+	cfg->irq_config_func();
```

Comment:
please split this change into 2 commits: one that makes driver multi-instance and 2nd pinctrl. If not possible for some reason please explain changes in more detail in the commit message

Reply:
ok, will split this commit into 2 commits